qualtran/bloqs/state_preparation/state_preparation_via_rotation.py

```diff
@@ -324,36 +390,53 @@ def signature(self) -> Signature:
             ]
         )
 
+    @property
+    def qrom_bloq(self) -> QROM:
+        return QROM(
+            (self.rom_values,),
+            selection_bitsizes=(self.selection_bitsize,),
+            target_bitsizes=(self.phase_bitsize,),
+            num_controls=self.control_bitsize,
+        )
+
+    @property
+    def add_into_phase_grad(self) -> AddIntoPhaseGrad:
+        return AddIntoPhaseGrad(self.phase_bitsize, self.phase_bitsize)
+
     def build_composite_bloq(self, bb: BloqBuilder, **soqs: SoquetT) -> Dict[str, SoquetT]:
         """Parameters:
         * control
         * selection (not necessary if selection_bitsize == 0)
         * phase_gradient
         """
-        qrom = QROM(
-            [np.array(self.rom_values)],
-            selection_bitsizes=(self.selection_bitsize,),
-            target_bitsizes=(self.phase_bitsize,),
-            num_controls=self.control_bitsize,
-        )
         # allocate a register to store the rotation angle
         soqs["target0_"] = bb.allocate(self.phase_bitsize)
         phase_grad = soqs.pop("phase_gradient")
         # load angles in rot_reg (line 1 of eq (8) in [1])
-        soqs = bb.add_d(qrom, **soqs)
+        soqs = bb.add_d(self.qrom_bloq, **soqs)
         # phase kickback via phase_grad += rot_reg (line 2 of eq (8) in [1])
         soqs["target0_"], phase_grad = bb.add(
-            AddIntoPhaseGrad(self.phase_bitsize, self.phase_bitsize),
-            x=soqs["target0_"],
-            phase_grad=phase_grad,
+            self.add_into_phase_grad, x=soqs["target0_"], phase_grad=phase_grad
         )
         # uncompute angle load in rot_reg to disentangle it from selection register
         # (line 1 of eq (8) in [1])
-        soqs = bb.add_d(qrom, **soqs)
+        soqs = bb.add_d(self.qrom_bloq.adjoint(), **soqs)
         soqs["phase_gradient"] = phase_grad
         bb.free(cast(Soquet, soqs.pop("target0_")))
         return soqs
 
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+        return {(self.qrom_bloq, 1), (self.qrom_bloq.adjoint(), 1), (self.add_into_phase_grad, 1)}
+
+    def __repr__(self):
```

Comment:
Why do we need a `__repr__` here?

Reply:
Nicer for debugging. We should probably add a `__repr__` everywhere we have classes that store `data: NDArray` as attributes (like qroms)